integrations/terraform/tfschema/custom_types.go

```diff
@@ -83,6 +92,37 @@ func CopyToBoolOption(diags diag.Diagnostics, o *apitypes.BoolOption, t attr.Typ
 	return value
 }
 
+func CopyFromNullBoolOption(diags diag.Diagnostics, tf attr.Value, o **apitypes.NullBoolOption) {
+	v, ok := tf.(types.Bool)
+	if !ok {
+		diags.AddError("Error reading from Terraform object", fmt.Sprintf("Can not convert %T to types.Bool", tf))
+		return
+	}
+
+	if v.IsNull() {
+		return
+	}
```

Comment:
This is the only functional difference between `BoolOption` and `NullBoolOption`